### PR TITLE
Enhancement: Performance enhancement for insert_rows method DbApiHook with fast executemany + SAP Hana support

### PR DESCRIPTION
**Pull request by dabla**
Added executemany parameter to insert_rows method of DbApiHook so the fast executemany method on the cursor can be used to achieve better performance when inserting in bulk instead of iterating over each row and inserting it one by one which is not efficient.  Also check if dialect is SAP Hana in _generate_insert_sql method so the UPSERT statement can be used as REPLACE INTO doesn't exist on SAP Hana.  Those changes have been unit tested in TestDbApiHook.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/microsoft/azure/transfers/local_to_adls.py [hunk 1]
```diff
@@ -96,3 +96,53 @@ def execute(self, context: Context) -> None:
             blocksize=self.blocksize,
             **self.extra_upload_options,
         )
+
+
+class DataToADLSOperator(BaseOperator):
```
Comment (eladkal) [inline airflow/providers/microsoft/azure/transfers/local_to_adls.py:101]: I don't understand how the azure changes are related to the `executemany` added in sql provider?
If these are two unrelated changes please split the PR.

Comment (dabla) [reply airflow/providers/microsoft/azure/transfers/local_to_adls.py:101]: I will close this one I think I started by mistake on same branch from other pull request that is still open
